newdle/api.py

```diff
@@ -237,3 +250,16 @@ def create_anonymous_participant(args, code):
     newdle.participants.add(participant)
     db.session.commit()
     return ParticipantSchema().jsonify(participant)
+
+
+@api.route('/newdle/<code>/participants/me', methods=('PUT',))
+def create_participant(code):
+    newdle = Newdle.query.filter_by(code=code).first_or_404('Invalid code')
+    user = g.user
+    name = user['first_name'] + ' ' + user['last_name']
+    participant = Participant.query.filter_by(auth_uid=user['uid']).first()
+    if not participant:
+        participant = Participant(name=name, email=user['email'], auth_uid=user['uid'])
+    newdle.participants.add(participant)
+    db.session.commit()
```

Comment:
these two lines should be inside the if body as well

Reply:
Aren't they going to be executed anyway if they are directly outside the if body? (There's no return inside if)

Reply:
but you don't need to execute them if you didn't create a new participant..

Reply:
Actually, the problem here is that I didn't query participants by newdle as well.